src/hdr_histogram.c

```diff
@@ -62,14 +63,34 @@ static void counts_inc_normalised(
     struct hdr_histogram* h, int32_t index, int64_t value)
 {
     int32_t normalised_index = normalize_index(h, index);
-    h->counts[normalised_index] += value;
-    h->total_count += value;
+    atomic_fetch_add_explicit(&h->counts[normalised_index], value, memory_order_relaxed);
+    atomic_fetch_add_explicit(&h->total_count, value, memory_order_relaxed);
 }
 
-static void update_min_max(struct hdr_histogram* h, int64_t value)
-{
-    h->min_value = (value < h->min_value && value != 0) ? value : h->min_value;
-    h->max_value = (value > h->max_value) ? value : h->max_value;
+static void update_min_max(struct hdr_histogram* h, int64_t value) {
+    bool updateValue = false;
+    if (value > 0) {
+        int64_t observedMin =
+                atomic_load_explicit(&h->min_value, memory_order_relaxed);
+
+        do {
+            updateValue = false;
+            if (value != 0 && value < observedMin) {
+                updateValue = true;
+            }
+        } while (updateValue && !atomic_compare_exchange_weak(
+                                        &h->min_value, &observedMin, value));
+    }
```

Comment:
Can be reduced to something like:

```
int64_t observedMin =
                atomic_load_explicit(&h->min_value, memory_order_relaxed);
while (value > 0 && value < observedMin && !atomic_compare_exchange_weak(&h->min_value, &observedMin, value)) {
}
```

Or 
```
if (value > 0) {
        int64_t observedMin = atomic_load_explicit(&h->min_value, memory_order_relaxed);
        while (value < observedMin && !atomic_compare_exchange_weak(&h->min_value, &observedMin, value)) {

        }     
    }
```

Similar for max

Reply:
Nit: You probably want the _explicit variant of compare_exchange_weak so you can specify relaxed memory ordering.

Reply:
Done